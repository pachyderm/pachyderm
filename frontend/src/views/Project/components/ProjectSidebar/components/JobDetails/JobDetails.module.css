.base {
  display: flex;
  flex-direction: column;
  flex: 1;
  overflow: hidden;
}

.heading {
  display: inline-flex;
  align-items: center;
  flex-wrap: wrap;
}

.jobLabel {
  margin-right: 0.25rem;
}

.headerSection {
  min-height: 4rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 0.5rem 1rem;
}

.seeMoreJobs {
  display: inline-block;
  align-items: center;
  padding-right: 0.25rem;
}

.pipelineSection {
  display: flex;
  flex: 1;
  border-top: 1px solid var(--line-grey);
  overflow: hidden;
}

.pipelineSpec {
  flex: 3;
  overflow: hidden;
  border-left: 1px solid var(--line-grey);
  min-width: 16rem;
}

.backArrow {
  transform: rotate(-90deg);
  margin-left: -0.875rem;
  margin-right: 0.3125rem;
}
