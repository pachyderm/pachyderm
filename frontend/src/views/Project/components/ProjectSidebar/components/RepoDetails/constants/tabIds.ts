export enum TAB_ID {
  INFO = 'info',
  COMMITS = 'commits',
}
