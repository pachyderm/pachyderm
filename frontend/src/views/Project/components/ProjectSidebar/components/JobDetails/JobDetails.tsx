import {JobState} from '@graphqlTypes';
import {Link, Tooltip, LoadingDots} from '@pachyderm/components';
import classnames from 'classnames';
import React from 'react';
import {NavLink, Redirect, Route} from 'react-router-dom';

import {useJobSet} from '@dash-frontend/hooks/useJobSet';
import useUrlState from '@dash-frontend/hooks/useUrlState';
import readableJobState from '@dash-frontend/lib/readableJobState';
import {PIPELINE_JOB_PATH} from '@dash-frontend/views/Project/constants/projectPaths';
import {
  jobsRoute,
  jobRoute,
  pipelineRoute,
} from '@dash-frontend/views/Project/utils/routes';

import InfoPanel from './components/InfoPanel';
import styles from './JobDetails.module.css';

type JobVisualState = 'BUSY' | 'ERROR' | 'SUCCESS';

const getVisualJobState = (state: JobState): JobVisualState => {
  switch (state) {
    case JobState.JOB_CREATED:
    case JobState.JOB_EGRESSING:
    case JobState.JOB_RUNNING:
    case JobState.JOB_STARTING:
      return 'BUSY';
    case JobState.JOB_FAILURE:
    case JobState.JOB_KILLED:
      return 'ERROR';
    default:
      return 'SUCCESS';
  }
};

const getJobStateHref = (state: JobVisualState) => {
  switch (state) {
    case 'BUSY':
      return '/dag_busy.svg';
    case 'ERROR':
      return '/dag_error.svg';
    default:
      return '/dag_success.svg';
  }
};

const JobDetails = () => {
  const {jobId, projectId, pipelineId} = useUrlState();
  const {jobSet, loading} = useJobSet({id: jobId, projectId});

  if (!pipelineId && jobSet && jobSet.jobs.length > 0) {
    return (
      <Redirect
        to={jobRoute({
          projectId,
          jobId,
          pipelineId: jobSet.jobs[0].pipelineName,
        })}
      />
    );
  }

  return (
    <div className={styles.base}>
      <section className={styles.headerSection}>
        <nav>
          <Link to={jobsRoute({projectId})} className={styles.seeMoreJobs}>
            All jobs{pipelineId ? ' >' : ''}
          </Link>
          {pipelineId && (
            <Link
              className={styles.seeMoreJobs}
              to={pipelineRoute({
                projectId,
                pipelineId,
                tabId: 'jobs',
              })}
            >
              Pipeline: {pipelineId}
            </Link>
          )}
        </nav>

        <h2 className={styles.heading}>Job {jobId}</h2>
      </section>

      <section className={styles.pipelineSection}>
        {loading && (
          <div
            className={styles.pipelineList}
            data-testid="JobDetails__loading"
          >
            <LoadingDots />
          </div>
        )}

        {!loading && (
          <nav>
            <ol className={styles.pipelineList}>
              {jobSet?.jobs.map((job) => {
                const jobVisualState = getVisualJobState(job.state);

                return (
                  <li key={job.pipelineName}>
                    <Tooltip
                      tooltipText={job.pipelineName}
                      tooltipKey={job.pipelineName}
                      size="large"
                      placement="right"
                    >
                      <NavLink
                        aria-current="true"
                        activeClassName={styles.active}
                        exact={true}
                        to={jobRoute({
                          projectId,
                          jobId,
                          pipelineId: job.pipelineName,
                        })}
                        className={classnames(styles.pipelineLink, {
                          [styles.error]: jobVisualState === 'ERROR',
                          [styles.success]: jobVisualState === 'SUCCESS',
                          [styles.busy]: jobVisualState === 'BUSY',
                        })}
                      >
                        <img
                          alt={`Pipeline job ${job.id} ${readableJobState(
                            job.state,
                          )}:`}
                          className={styles.pipelineIcon}
                          src={getJobStateHref(jobVisualState)}
                        />
                        <div className={styles.pipelineLinkText}>
                          {job.pipelineName}
                        </div>
                      </NavLink>
                    </Tooltip>
                  </li>
                );
              })}
            </ol>
          </nav>
        )}

        <div className={styles.pipelineSpec}>
          <Route path={PIPELINE_JOB_PATH}>
            <InfoPanel />
          </Route>
        </div>
      </section>
    </div>
  );
};

export default JobDetails;
