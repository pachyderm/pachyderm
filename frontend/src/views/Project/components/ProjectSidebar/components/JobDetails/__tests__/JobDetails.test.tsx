import {render, waitFor, within, screen} from '@testing-library/react';
import {fromUnixTime, formatDistanceToNowStrict} from 'date-fns';
import React from 'react';
import {Route} from 'react-router';

import {
  withContextProviders,
  click,
  getUrlState,
} from '@dash-frontend/testHelpers';
import {PROJECT_JOB_PATH} from '@dash-frontend/views/Project/constants/projectPaths';
import {
  fileBrowserRoute,
  jobRoute,
} from '@dash-frontend/views/Project/utils/routes';

import JobDetails from '..';

describe('Job Details', () => {
  const TestBed = withContextProviders(() => {
    return <Route path={PROJECT_JOB_PATH} component={JobDetails} />;
  });

  const projectId = 'Data-Cleaning-Process';
  const jobId = '23b9af7d5d4343219bc8e02ff4acd33a';

  beforeEach(() => {
    window.history.replaceState('', '', jobRoute({projectId, jobId}));
  });

  it('should display the list of pipelines that produced jobs in the set', async () => {
    render(<TestBed />);

    await waitFor(() =>
      expect(
        screen.queryByTestId('JobDetails__loading'),
      ).not.toBeInTheDocument(),
    );
    await waitFor(() =>
      expect(
        screen.queryByTestId('Description__InputsSkeleton'),
      ).not.toBeInTheDocument(),
    );

    const withinNavList = within(screen.getByRole('list'));

    const likelihoodsLink = withinNavList.getByRole('link', {
      name: `likelihoods`,
    });
    expect(likelihoodsLink).toBeInTheDocument();
    expect(likelihoodsLink).toHaveAttribute(
      'href',
      decodeURI(jobRoute({projectId, jobId, pipelineId: 'likelihoods'}, false)),
    );

    const modelsLink = withinNavList.getByRole('link', {
      name: `models`,
    });
    expect(modelsLink).toBeInTheDocument();
    expect(modelsLink).toHaveAttribute(
      'href',
      decodeURI(jobRoute({projectId, jobId, pipelineId: 'models'}, false)),
    );

    const jointCallLink = withinNavList.getByRole('link', {
      name: `joint_call`,
    });
    expect(jointCallLink).toBeInTheDocument();
    expect(jointCallLink).toHaveAttribute(
      'href',
      decodeURI(jobRoute({projectId, jobId, pipelineId: 'joint_call'}, false)),
    );

    const splitLink = withinNavList.getByRole('link', {
      name: `split`,
    });
    expect(splitLink).toBeInTheDocument();
    expect(splitLink).toHaveAttribute(
      'href',
      decodeURI(jobRoute({projectId, jobId, pipelineId: 'split'}, false)),
    );

    const testLink = withinNavList.getByRole('link', {
      name: `test`,
    });
    expect(testLink).toBeInTheDocument();
    expect(testLink).toHaveAttribute(
      'href',
      decodeURI(jobRoute({projectId, jobId, pipelineId: 'test'}, false)),
    );

    await click(testLink);

    await waitFor(() =>
      expect(
        screen.queryByTestId('Description__InputsSkeleton'),
      ).not.toBeInTheDocument(),
    );

    expect(window.location.pathname).toBe(
      jobRoute({projectId, jobId, pipelineId: 'test'}, false),
    );
  });

  it('should default to the first job in the set if not specified', async () => {
    render(<TestBed />);

    await waitFor(() =>
      expect(
        screen.queryByTestId('JobDetails__loading'),
      ).not.toBeInTheDocument(),
    );
    await waitFor(() =>
      expect(
        screen.queryByTestId('Description__InputsSkeleton'),
      ).not.toBeInTheDocument(),
    );

    expect(window.location.pathname).toBe(
      jobRoute({projectId, jobId, pipelineId: 'likelihoods'}, false),
    );
    await waitFor(() =>
      expect(
        screen.queryByTestId('InfoPanel__loading'),
      ).not.toBeInTheDocument(),
    );

    expect(screen.getByTestId('InfoPanel__pipeline')).toHaveTextContent(
      'likelihoods',
    );
    expect(screen.getByTestId('InfoPanel__state')).toHaveTextContent('Failure');
    expect(screen.getByTestId('RuntimeStats__started')).toHaveTextContent(
      formatDistanceToNowStrict(fromUnixTime(1614136190), {addSuffix: true}),
    );
    expect(screen.getByTestId('InfoPanel__success')).toHaveTextContent(/^0$/);
    expect(screen.getByTestId('InfoPanel__failed')).toHaveTextContent(/^100$/);
    expect(screen.getByTestId('InfoPanel__skipped')).toHaveTextContent(/^0$/);
    expect(screen.getByTestId('InfoPanel__recovered')).toHaveTextContent(/^0$/);
    expect(screen.getByTestId('InfoPanel__total')).toHaveTextContent(/^100$/);
  });

  it('should display correct pipeline job based on url', async () => {
    window.history.replaceState(
      '',
      '',
      jobRoute({projectId, jobId, pipelineId: 'models'}),
    );

    render(<TestBed />);

    await waitFor(() =>
      expect(
        screen.queryByTestId('JobDetails__loading'),
      ).not.toBeInTheDocument(),
    );
    await waitFor(() =>
      expect(
        screen.queryByTestId('Description__InputsSkeleton'),
      ).not.toBeInTheDocument(),
    );

    const pipelineName = await screen.findByTestId('InfoPanel__pipeline');
    expect(pipelineName).toHaveTextContent('models');
  });

  it('should display job durations', async () => {
    window.history.replaceState(
      '',
      '',
      jobRoute({
        projectId: 'Solar-Panel-Data-Sorting',
        jobId: '33b9af7d5d4343219bc8e02ff44cd55a',
        pipelineId: 'montage',
      }),
    );

    render(<TestBed />);

    await waitFor(() =>
      expect(
        screen.queryByTestId('JobDetails__loading'),
      ).not.toBeInTheDocument(),
    );
    await waitFor(() =>
      expect(
        screen.queryByTestId('Description__InputsSkeleton'),
      ).not.toBeInTheDocument(),
    );
    await click(await screen.findByTestId('RuntimeStats__duration'));

    expect(
      screen.queryByTestId('RuntimeStats__durationDetails'),
    ).toMatchSnapshot();
  });

  it('should display job failure', async () => {
    window.history.replaceState(
      '',
      '',
      jobRoute({
        projectId,
        jobId,
        pipelineId: 'likelihoods',
      }),
    );

    render(<TestBed />);

    await waitFor(() =>
      expect(
        screen.queryByTestId('JobDetails__loading'),
      ).not.toBeInTheDocument(),
    );
    await waitFor(() =>
      expect(
        screen.queryByTestId('Description__InputsSkeleton'),
      ).not.toBeInTheDocument(),
    );
    expect(await screen.findByTestId('InfoPanel__state')).toHaveTextContent(
      'Failure',
    );
    expect(await screen.findByTestId('InfoPanel__reason')).toHaveTextContent(
      'failed',
    );
  });

  it('should correctly render extra details in JSON blob', async () => {
    window.history.replaceState(
      '',
      '',
      jobRoute({
        projectId: 'Solar-Panel-Data-Sorting',
        jobId: '33b9af7d5d4343219bc8e02ff44cd55a',
        pipelineId: 'montage',
      }),
    );

    render(<TestBed />);

    await waitFor(() =>
      expect(
        screen.queryByTestId('JobDetails__loading'),
      ).not.toBeInTheDocument(),
    );
    await waitFor(() =>
      expect(
        screen.queryByTestId('Description__InputsSkeleton'),
      ).not.toBeInTheDocument(),
    );

    expect(screen.queryByTestId('InfoPanel__details')).toMatchSnapshot();
  });

  it('should fallback to first job if pipeline job cannot be found', async () => {
    window.history.replaceState(
      '',
      '',
      jobRoute({projectId, jobId, pipelineId: 'bogus'}),
    );

    render(<TestBed />);

    await waitFor(() =>
      expect(
        screen.queryByTestId('JobDetails__loading'),
      ).not.toBeInTheDocument(),
    );
    await waitFor(() =>
      expect(
        screen.queryByTestId('Description__InputsSkeleton'),
      ).not.toBeInTheDocument(),
    );

    expect(window.location.pathname).toBe(
      jobRoute({projectId, jobId, pipelineId: 'likelihoods'}, false),
    );
  });

  it('should allow the user to navigate to the output commit', async () => {
    window.history.replaceState(
      '',
      '',
      jobRoute({projectId, jobId, pipelineId: 'models'}),
    );

    render(<TestBed />);

    await waitFor(() =>
      expect(
        screen.queryByTestId('JobDetails__loading'),
      ).not.toBeInTheDocument(),
    );
    await waitFor(() =>
      expect(
        screen.queryByTestId('Description__InputsSkeleton'),
      ).not.toBeInTheDocument(),
    );
    await screen.findByTestId('InfoPanel__commitLink');

    const outputCommitLink = screen.getByTestId('InfoPanel__commitLink');

    await click(outputCommitLink);

    expect(window.location.pathname).toBe(
      fileBrowserRoute(
        {
          repoId: 'models',
          branchId: 'master',
          commitId: jobId,
          projectId,
        },
        false,
      ),
    );

    expect(getUrlState()).toMatchObject({
      prevFileBrowserPath: decodeURI(
        jobRoute({projectId, jobId, pipelineId: 'models'}, false),
      ),
    });
  });
});
