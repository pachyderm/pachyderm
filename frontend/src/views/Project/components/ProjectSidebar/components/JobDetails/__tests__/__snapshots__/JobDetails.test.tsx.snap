// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Details should correctly render extra details in JSON blob 1`] = `
<div
  data-testid="InfoPanel__details"
>
  <div
    class="header"
  >
    <h5>
      Job Definition
    </h5>
    <div
      class="settings _base_1p360_1 _horizontal_1p360_45 _spacing16_1p360_21"
    >
      <div
        class="_base_11ori_1"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          class="_base_1o7uy_1 _hideChevron_1o7uy_33 _purple_1o7uy_42"
          data-testid="DropdownButton__button"
          type="button"
        >
          <span
            class="_children_1o7uy_26"
          >
            <div
              class="_base_19s37_1 _plum_19s37_22"
            >
              <svg
                height="20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                >
                  <path
                    d="M0 0h20v20H0z"
                  />
                  <path
                    d="M10.794 1a.75.75 0 0 1 .744.658l.164 1.3a.19.19 0 0 0 .138.158 7.045 7.045 0 0 1 1.725.718.188.188 0 0 0 .21-.014l1.034-.808a.75.75 0 0 1 .99.062L16.926 4.2a.75.75 0 0 1 .061.991l-.806 1.037a.188.188 0 0 0-.014.21c.315.54.557 1.12.719 1.724.02.074.082.128.157.138l1.301.162a.75.75 0 0 1 .657.744v1.588a.75.75 0 0 1-.657.746l-1.3.163a.186.186 0 0 0-.158.137 7.048 7.048 0 0 1-.72 1.725.188.188 0 0 0 .015.209l.808 1.038a.75.75 0 0 1-.061.99l-1.125 1.125a.75.75 0 0 1-.99.061l-1.037-.806a.186.186 0 0 0-.21-.014 7.03 7.03 0 0 1-1.725.718.189.189 0 0 0-.138.158l-.162 1.3a.75.75 0 0 1-.744.657h-1.59a.75.75 0 0 1-.744-.657l-.163-1.3a.189.189 0 0 0-.138-.158 7.03 7.03 0 0 1-1.725-.718.188.188 0 0 0-.209.014l-1.037.806a.75.75 0 0 1-.99-.062L3.074 15.8a.75.75 0 0 1-.061-.99l.806-1.04a.188.188 0 0 0 .014-.209 7.048 7.048 0 0 1-.719-1.724.186.186 0 0 0-.158-.137l-1.3-.163A.75.75 0 0 1 1 10.793V9.205a.75.75 0 0 1 .658-.744l1.3-.164a.187.187 0 0 0 .157-.137 7.048 7.048 0 0 1 .72-1.725.188.188 0 0 0-.015-.21l-.809-1.034a.75.75 0 0 1 .062-.991l1.125-1.125a.75.75 0 0 1 .99-.062l1.039.807c.06.047.143.053.209.014a7.045 7.045 0 0 1 1.725-.718.189.189 0 0 0 .138-.158l.163-1.301A.75.75 0 0 1 9.206 1ZM10 5.5a4.5 4.5 0 1 0 0 9 4.5 4.5 0 0 0 0-9Z"
                    fill="currentcolor"
                    fill-rule="nonzero"
                  />
                </g>
              </svg>
            </div>
          </span>
        </button>
        <div
          aria-hidden="true"
          class="_base_1kfrl_57 _right_1kfrl_77"
          role="menu"
        >
          <button
            class="_base_bt078_1"
            data-testid="DropdownMenuItem__button"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            Copy
          </button>
          <button
            class="_base_bt078_1"
            data-testid="DropdownMenuItem__button"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            Download JSON
          </button>
          <button
            class="_base_bt078_1"
            data-testid="DropdownMenuItem__button"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            Download YAML
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="codeBody"
  >
    <div
      class="codeSection"
    >
      <div>
        <div
          class="code language-yaml depth0 hljs"
        />
        <div
          class="header"
        >
          <div
            class="_base_19s37_1 chevron undefined _small_19s37_54"
          >
            <svg
              height="20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
              >
                <path
                  d="M0 0h20v20H0z"
                />
                <path
                  d="M8.94 13.841a1.5 1.5 0 0 0 2.007.103l.114-.103 6-6-2.122-2.121L10 10.66 5.06 5.72 2.94 7.84l6 6Z"
                  fill="currentcolor"
                  fill-rule="nonzero"
                />
              </g>
            </svg>
          </div>
          <div
            class="code language-yaml depth0 hljs"
          >
            <span
              class="hljs-attr"
            >
              input:
            </span>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class="codeSection"
          >
            <div>
              <div
                class="code language-yaml depth1 hljs"
              >
                <span
                  class="hljs-attr"
                >
                  joinList:
                </span>
                 []

                <span
                  class="hljs-attr"
                >
                  groupList:
                </span>
                 []

              </div>
              <div
                class="header"
              >
                <div
                  class="_base_19s37_1 chevron undefined _small_19s37_54"
                >
                  <svg
                    height="20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <path
                        d="M0 0h20v20H0z"
                      />
                      <path
                        d="M8.94 13.841a1.5 1.5 0 0 0 2.007.103l.114-.103 6-6-2.122-2.121L10 10.66 5.06 5.72 2.94 7.84l6 6Z"
                        fill="currentcolor"
                        fill-rule="nonzero"
                      />
                    </g>
                  </svg>
                </div>
                <div
                  class="code language-yaml depth1 hljs"
                >
                  <span
                    class="hljs-attr"
                  >
                    crossList:
                  </span>
                </div>
              </div>
              <div
                class=""
              >
                <div
                  class="codeSection"
                >
                  <div
                    class="code language-yaml depth2 hljs"
                  >
                    <span
                      class="hljs-bullet"
                    >
                      -
                    </span>
                     
                    <span
                      class="hljs-attr"
                    >
                      pfs:
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      name:
                    </span>
                     
                    <span
                      class="hljs-string"
                    >
                      edges
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      repo:
                    </span>
                     
                    <span
                      class="hljs-string"
                    >
                      edges
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      repoType:
                    </span>
                     
                    <span
                      class="hljs-string"
                    >
                      ''
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      branch:
                    </span>
                     
                    <span
                      class="hljs-string"
                    >
                      master
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      commit:
                    </span>
                     
                    <span
                      class="hljs-string"
                    >
                      ''
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      glob:
                    </span>
                     
                    <span
                      class="hljs-string"
                    >
                      ''
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      joinOn:
                    </span>
                     
                    <span
                      class="hljs-string"
                    >
                      ''
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      outerJoin:
                    </span>
                     
                    <span
                      class="hljs-literal"
                    >
                      false
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      groupBy:
                    </span>
                     
                    <span
                      class="hljs-string"
                    >
                      ''
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      lazy:
                    </span>
                     
                    <span
                      class="hljs-literal"
                    >
                      false
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      emptyFiles:
                    </span>
                     
                    <span
                      class="hljs-literal"
                    >
                      false
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      s3:
                    </span>
                     
                    <span
                      class="hljs-literal"
                    >
                      false
                    </span>
                    
  
                    <span
                      class="hljs-attr"
                    >
                      joinList:
                    </span>
                     []
  
                    <span
                      class="hljs-attr"
                    >
                      groupList:
                    </span>
                     []
  
                    <span
                      class="hljs-attr"
                    >
                      crossList:
                    </span>
                     []
  
                    <span
                      class="hljs-attr"
                    >
                      unionList:
                    </span>
                     []

                    <span
                      class="hljs-bullet"
                    >
                      -
                    </span>
                     
                    <span
                      class="hljs-attr"
                    >
                      pfs:
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      name:
                    </span>
                     
                    <span
                      class="hljs-string"
                    >
                      images
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      repo:
                    </span>
                     
                    <span
                      class="hljs-string"
                    >
                      images
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      repoType:
                    </span>
                     
                    <span
                      class="hljs-string"
                    >
                      ''
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      branch:
                    </span>
                     
                    <span
                      class="hljs-string"
                    >
                      master
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      commit:
                    </span>
                     
                    <span
                      class="hljs-string"
                    >
                      ''
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      glob:
                    </span>
                     
                    <span
                      class="hljs-string"
                    >
                      ''
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      joinOn:
                    </span>
                     
                    <span
                      class="hljs-string"
                    >
                      ''
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      outerJoin:
                    </span>
                     
                    <span
                      class="hljs-literal"
                    >
                      false
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      groupBy:
                    </span>
                     
                    <span
                      class="hljs-string"
                    >
                      ''
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      lazy:
                    </span>
                     
                    <span
                      class="hljs-literal"
                    >
                      false
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      emptyFiles:
                    </span>
                     
                    <span
                      class="hljs-literal"
                    >
                      false
                    </span>
                    
    
                    <span
                      class="hljs-attr"
                    >
                      s3:
                    </span>
                     
                    <span
                      class="hljs-literal"
                    >
                      false
                    </span>
                    
  
                    <span
                      class="hljs-attr"
                    >
                      joinList:
                    </span>
                     []
  
                    <span
                      class="hljs-attr"
                    >
                      groupList:
                    </span>
                     []
  
                    <span
                      class="hljs-attr"
                    >
                      crossList:
                    </span>
                     []
  
                    <span
                      class="hljs-attr"
                    >
                      unionList:
                    </span>
                     []

                  </div>
                </div>
              </div>
            </div>
            <div
              class="code language-yaml depth1 hljs"
            >
              <span
                class="hljs-attr"
              >
                unionList:
              </span>
               []

            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="code language-yaml depth0 hljs"
        >
          <span
            class="hljs-attr"
          >
            transform:
          </span>
           {}

        </div>
        <div
          class="header"
        >
          <div
            class="_base_19s37_1 chevron undefined _small_19s37_54"
          >
            <svg
              height="20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
              >
                <path
                  d="M0 0h20v20H0z"
                />
                <path
                  d="M8.94 13.841a1.5 1.5 0 0 0 2.007.103l.114-.103 6-6-2.122-2.121L10 10.66 5.06 5.72 2.94 7.84l6 6Z"
                  fill="currentcolor"
                  fill-rule="nonzero"
                />
              </g>
            </svg>
          </div>
          <div
            class="code language-yaml depth0 hljs"
          >
            <span
              class="hljs-attr"
            >
              details:
            </span>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class="codeSection"
          >
            <div>
              <div
                class="code language-yaml depth1 hljs"
              >
                <span
                  class="hljs-attr"
                >
                  pipelineVersion:
                </span>
                 
                <span
                  class="hljs-number"
                >
                  1
                </span>
                

                <span
                  class="hljs-attr"
                >
                  dataTotal:
                </span>
                 
                <span
                  class="hljs-number"
                >
                  5
                </span>
                

                <span
                  class="hljs-attr"
                >
                  dataFailed:
                </span>
                 
                <span
                  class="hljs-number"
                >
                  4
                </span>
                

              </div>
              <div
                class="header"
              >
                <div
                  class="_base_19s37_1 chevron undefined _small_19s37_54"
                >
                  <svg
                    height="20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <path
                        d="M0 0h20v20H0z"
                      />
                      <path
                        d="M8.94 13.841a1.5 1.5 0 0 0 2.007.103l.114-.103 6-6-2.122-2.121L10 10.66 5.06 5.72 2.94 7.84l6 6Z"
                        fill="currentcolor"
                        fill-rule="nonzero"
                      />
                    </g>
                  </svg>
                </div>
                <div
                  class="code language-yaml depth1 hljs"
                >
                  <span
                    class="hljs-attr"
                  >
                    stats:
                  </span>
                </div>
              </div>
              <div
                class=""
              >
                <div
                  class="codeSection"
                >
                  <div>
                    <div
                      class="code language-yaml depth2 hljs"
                    />
                    <div
                      class="header"
                    >
                      <div
                        class="_base_19s37_1 chevron undefined _small_19s37_54"
                      >
                        <svg
                          height="20"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fill-rule="evenodd"
                          >
                            <path
                              d="M0 0h20v20H0z"
                            />
                            <path
                              d="M8.94 13.841a1.5 1.5 0 0 0 2.007.103l.114-.103 6-6-2.122-2.121L10 10.66 5.06 5.72 2.94 7.84l6 6Z"
                              fill="currentcolor"
                              fill-rule="nonzero"
                            />
                          </g>
                        </svg>
                      </div>
                      <div
                        class="code language-yaml depth2 hljs"
                      >
                        <span
                          class="hljs-attr"
                        >
                          downloadTime:
                        </span>
                      </div>
                    </div>
                    <div
                      class=""
                    >
                      <div
                        class="codeSection"
                      >
                        <div
                          class="code language-yaml depth3 hljs"
                        >
                          <span
                            class="hljs-attr"
                          >
                            seconds:
                          </span>
                           
                          <span
                            class="hljs-number"
                          >
                            10
                          </span>
                          

                          <span
                            class="hljs-attr"
                          >
                            nanos:
                          </span>
                           
                          <span
                            class="hljs-number"
                          >
                            269391100
                          </span>
                          

                        </div>
                      </div>
                    </div>
                  </div>
                  <div>
                    <div
                      class="code language-yaml depth2 hljs"
                    />
                    <div
                      class="header"
                    >
                      <div
                        class="_base_19s37_1 chevron undefined _small_19s37_54"
                      >
                        <svg
                          height="20"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fill-rule="evenodd"
                          >
                            <path
                              d="M0 0h20v20H0z"
                            />
                            <path
                              d="M8.94 13.841a1.5 1.5 0 0 0 2.007.103l.114-.103 6-6-2.122-2.121L10 10.66 5.06 5.72 2.94 7.84l6 6Z"
                              fill="currentcolor"
                              fill-rule="nonzero"
                            />
                          </g>
                        </svg>
                      </div>
                      <div
                        class="code language-yaml depth2 hljs"
                      >
                        <span
                          class="hljs-attr"
                        >
                          processTime:
                        </span>
                      </div>
                    </div>
                    <div
                      class=""
                    >
                      <div
                        class="codeSection"
                      >
                        <div
                          class="code language-yaml depth3 hljs"
                        >
                          <span
                            class="hljs-attr"
                          >
                            seconds:
                          </span>
                           
                          <span
                            class="hljs-number"
                          >
                            20
                          </span>
                          

                          <span
                            class="hljs-attr"
                          >
                            nanos:
                          </span>
                           
                          <span
                            class="hljs-number"
                          >
                            531186700
                          </span>
                          

                        </div>
                      </div>
                    </div>
                  </div>
                  <div>
                    <div
                      class="code language-yaml depth2 hljs"
                    />
                    <div
                      class="header"
                    >
                      <div
                        class="_base_19s37_1 chevron undefined _small_19s37_54"
                      >
                        <svg
                          height="20"
                          width="20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fill-rule="evenodd"
                          >
                            <path
                              d="M0 0h20v20H0z"
                            />
                            <path
                              d="M8.94 13.841a1.5 1.5 0 0 0 2.007.103l.114-.103 6-6-2.122-2.121L10 10.66 5.06 5.72 2.94 7.84l6 6Z"
                              fill="currentcolor"
                              fill-rule="nonzero"
                            />
                          </g>
                        </svg>
                      </div>
                      <div
                        class="code language-yaml depth2 hljs"
                      >
                        <span
                          class="hljs-attr"
                        >
                          uploadTime:
                        </span>
                      </div>
                    </div>
                    <div
                      class=""
                    >
                      <div
                        class="codeSection"
                      >
                        <div
                          class="code language-yaml depth3 hljs"
                        >
                          <span
                            class="hljs-attr"
                          >
                            seconds:
                          </span>
                           
                          <span
                            class="hljs-number"
                          >
                            30
                          </span>
                          

                          <span
                            class="hljs-attr"
                          >
                            nanos:
                          </span>
                           
                          <span
                            class="hljs-number"
                          >
                            231186700
                          </span>
                          

                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="code language-yaml depth2 hljs"
                  >
                    <span
                      class="hljs-attr"
                    >
                      downloadBytes:
                    </span>
                     
                    <span
                      class="hljs-number"
                    >
                      2896
                    </span>
                    

                  </div>
                </div>
              </div>
            </div>
            <div
              class="code language-yaml depth1 hljs"
            >
              <span
                class="hljs-attr"
              >
                salt:
              </span>
               
              <span
                class="hljs-string"
              >
                d5631d7df40d4b1195bc46f1f146d6a5
              </span>
              

              <span
                class="hljs-attr"
              >
                datumTries:
              </span>
               
              <span
                class="hljs-number"
              >
                3
              </span>
              

            </div>
          </div>
        </div>
      </div>
      <div
        class="code language-yaml depth0 hljs"
      />
    </div>
  </div>
</div>
`;
